django_netjsonconfig/base/admin.py

```diff
@@ -200,6 +205,15 @@ def download_view(self, request, pk):
         return send_file(filename='{0}.tar.gz'.format(config.name),
                          contents=config_archive.getvalue())
 
+    def device_context_view(self, request, pk):
+        device = get_object_or_404(self.model, pk=pk)
+        if hasattr(device, 'config'):
+            context = json.dumps(device.config.get_context())
+        else:
+            device_config = Config(device=device)
+            context = json.dumps(device_config.get_context())
```

Comment:
please do not duplicate code.
Use `device_config` to store the right config instance and then write only one `json.dumps(device_config.get_context())` call.